Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -1435,3 +1435,118 @@ extension HTTPMethod: RawRepresentable {
         }
     }
 }
+
+extension HTTPResponseHead {
+    public var hasContentLength: Bool {
+        return headers.hasContentLength
+    }
+
+    public var contentLength: Int? {
+        return headers.contentLength
+    }
+}
+
+extension HTTPRequestHead {
+    // Returns {@code true} if and only if the specified message contains an expect header and the only expectation
+    // present is the 100-continue expectation. Note that this method returns {@code false} if the expect header is
+    // not valid for the message (e.g., the version on the message is HTTP/1.0).
+    public var isContinueExpected: Bool {
+        return headers.isContinueExpected(version: version)
+    }
+
+    // Returns {@code true} if the specified message contains an expect header specifying an expectation that is not
+    // supported. Note that this method returns {@code false} if the expect header is not valid for the message
+    // (e.g., the version on the message is HTTP/1.0).
+    public var isUnsupportedExpectation: Bool {
+        return headers.isUnsupportedExpectation(version: version)
+    }
+
+    public var hasContentLength: Bool {
+        return headers.hasContentLength
+    }
+
+    public var contentLength: Int? {
+        return headers.contentLength
+    }
+}
+
+extension HTTPHeaders {
+    var hasContentLength: Bool {
+        return self.contains(name: "content-length") && self[canonicalForm: "content-length"].count == 1
+    }
+
+    var contentLength: Int? {
+        return self.hasContentLength ? Int(self.first(name: "content-length") ?? "") : nil
```

Comment:
This also loops redundantly: we don't need to preflight whether `hasContentLength` is true, as we'll answer that by using `self.first(name:)` anyway. Also, let's not ask Swift to try to work out if `""` is a valid `Int`:

```swift
self.first(name: "content-length").flatMap { Int($0) }
```

is just as good a spelling here.